src/TitleDetails/TitleDetails.js

```diff
@@ -224,7 +248,15 @@ const TitleDetails = ({
           onCancel={toggleAcknowledgeNote}
           onConfirm={() => {
             toggleAcknowledgeNote();
-            toggleAddPieceModal();
+
+            if (clickedButton === EXPECTED_PIECES_BUTTONS.addPiece) {
```

Comment:
that's might look like antipattern. You might want to think about passing the right callback instead of `clickedButton`

Reply:
But `ConfirmationModal` is common, so I need to know somehow what button was clicked. Any ideas?

Reply:
As I wrote above, the whole solution of keeping leads of what button user clicked and making decisions basing on these leads - is kind of code smell. There are more simple solutions, which use interface segregation and low coupling. 
_At least, it's javascript, everything is a callback (just joking)_